### PR TITLE
Draft for feedback on code removal to simplify Web Access work

### PR DESCRIPTION
**Pull request by jnm2**
@KirillOsenkov This is just a draft so I can get your feedback on the code removal. I deleted an unused public method but I'm not sure if you're okay with that. (Is HtmlGenerator used as public API, or is this deleting work that you'd like to build on in the future?) The motivation for the deletes was to simplify the parts of GetWebLink that I can't test to see if they're working, when I get to the point of unifying GetWebLink for source files and XML/TS files.

### CODE REVIEW

File: src/HtmlGenerator/Pass1-Generation/DocumentGenerator.cs [hunk 1]
```diff
@@ -321,22 +321,7 @@ private string GetWebLink()
...
-
-            string url = string.Format(
-                urlTemplate,
-                serverPath,
-                filePath);
-            return url;
+            return null;
         }
 
         private string GetDocumentPathFromSourceSolutionRoot()
```
Comment (jnm2) [inline src/HtmlGenerator/Pass1-Generation/DocumentGenerator.cs:327]: This method has one remaining reference.

File: src/HtmlGenerator/Pass1-Generation/GenerateFromBuildLog.cs [hunk 2]
```diff
@@ -16,34 +16,6 @@ public class GenerateFromBuildLog
...
-                {
-                    logFile
-                });
-
-            var buildlogInvocations = invocations.Select(i => new CompilerInvocation()
-            {
-                CommandLineArguments = i.CommandLine,
-                OutputAssemblyPath = i.OutputAssemblyPath,
-                ProjectFilePath = i.ProjectFilePath,
-                ServerPath = serverPath,
```
Comment (jnm2) [inline src/HtmlGenerator/Pass1-Generation/GenerateFromBuildLog.cs:38]: This was the only method in which a non-blank value could have ended up in SolutionGenerator.ServerPath.

Comment (jnm2) [reply src/HtmlGenerator/Pass1-Generation/GenerateFromBuildLog.cs:38]: Another option would be to remove the optional parameter and leave the rest of the method.

File: src/HtmlGenerator/Pass1-Generation/GenerateFromBuildLog.cs [hunk 3]
```diff
@@ -16,34 +16,6 @@ public class GenerateFromBuildLog
         public static readonly Dictionary<string, string> AssemblyNameToFilePathMap =
             new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
-        public static void IndexLogFile(string logFile, string serverPath = null, string solutionRoot = null)
```
Comment (KirillOsenkov) [inline src/HtmlGenerator/Pass1-Generation/GenerateFromBuildLog.cs:19]: Yes, this is dead and can go. This was parsing MSBuild text log files before binlogs were around.

**Comment by KirillOsenkov** [comment]
Looks OK.

**Comment by jnm2** [comment]
Thanks!

**Comment by KirillOsenkov** [comment]
If you're in the mood to delete code, this entire project can go now:
https://github.com/KirillOsenkov/SourceBrowser/tree/main/src/BuildLogParser

I've spent a lot of time on this in 2012-2013 to fine tune this logic. It was manually parsing MSBuild text log files, extracting compiler invocations so I could build a workspace, for scenarios where MSBuildWorkspace didn't work. It was so painful to get right, error prone and fragile. Check out https://github.com/KirillOsenkov/SourceBrowser/blob/main/src/BuildLogParser/LogAnalyzer.cs for a history lesson if interested.

Now we have binlogs so all of that can just die in a fire.

**Comment by jnm2** [comment]
Always, I love deleting code!

This method is unused. Do you still want it?
https://github.com/KirillOsenkov/SourceBrowser/blob/21ad7b7b86f3a5cdb68cfb6b5e46873422d5341b/src/HtmlGenerator/Pass1-Generation/GenerateFromBuildLog.cs#L110-L126

**Pull request closed without merging**